File: 2-Regression/4-Simple Linear Regression/simple_linear_regression_risk.py
# Simple Linear Regression

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values # -1 removes last col, makes matrix of features / independent variables
y = dataset.iloc[:, 1].values # 1 selects only 2nd col for dependent variable vector

# Splitting the dataset into the Training set and Test set, make test size a third 
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# Feature Scaling, comment out because not needed for simple linear regression, as library takes care of it.
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)"""

# Fitting Simple Linear Regression to the Training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression() # No params needed
regressor.fit(X_train, y_train) # Fits regressor object to the training sets, regressor is machine that learns on training set, can use this to make predictions on test set

# Predicting the Test set results
# Put predictions in vector
y_pred = regressor.predict(X_test)
# compare y_test (actual salaries) with y_pred

File: 2-Regression/4-Simple Linear Regression/simple_linear_regression_risk.R
# Simple Linear Regression

# Importing the dataset
dataset = read.csv('Salary_Data.csv')

# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3) # 2/3 in Training Set
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Feature Scaling is taken care of by linear regression package in R
# training_set = scale(training_set)
# test_set = scale(test_set)

# data preprocessing complete

# Fitting Simple Linear Regression to the Training set
regressor = lm(formula = Salary ~ YearsExperience, 
               data = training_set)

# Type summary(regressor) in console and notice *** for strength of dependence
# Low P value indicates independent variable is highly significant

# Predicting the Test set Results
y_pred = predict(regressor, newdata = test_set)
# After execution, type y_pred in console to see values